iOS/Objective C/IronSourceDemoApp/NativeAdViewController.m

```diff
@@ -0,0 +1,158 @@
+        if ([pair.lastObject isEqual:currentAdView]) {
+            index = i;
+            break;
+        }
+    }
+    
+    if (index != -1) {
+        NSArray *pair = self.nativeAdlist[index];
+        LevelPlayNativeAd *nativeAd = pair[0];
+        NativeAdView *adView = pair[1];
+        
+        [nativeAd destroyAd];
+        [self.nativeAdlist removeObject:pair];
+        
+        NSIndexPath *indexPath = [NSIndexPath indexPathForRow:index inSection:0];
+        [self.tableViewData removeObject:adView];
+        [self.tableView deleteRowsAtIndexPaths:@[indexPath] withRowAnimation:UITableViewRowAnimationAutomatic];
+    }
+}
+
+#pragma mark - LevelPlayNativeAdDelegate Functions
+
+// ISNativeAdLoadDelegate
+
+// This method gets invoked after a failed attempt to load Native Ad.
+// If it does happen, check out 'error' for more information and consult our
+// Knowledge center.
+- (void)didFailToLoad:(nonnull LevelPlayNativeAd *)nativeAd withError:(nonnull NSError *)error {
+    NSLog(@"%s",__PRETTY_FUNCTION__);
+}
+
+// This method gets invoked after a native ad was loaded.
+// Then the information will be passed to the NativeAdView and added to your data source.
+- (void)didLoad:(nonnull LevelPlayNativeAd *)nativeAd withAdInfo:(nonnull ISAdInfo *)adInfo {
+    NSLog(@"%s",__PRETTY_FUNCTION__);
+    
+    NSInteger index = -1;
+    for (NSInteger i = 0; i < self.nativeAdlist.count; i++) {
+        NSArray *pair = self.nativeAdlist[i];
+        if ([pair.firstObject isEqual:nativeAd]) {
+            index = i;
+            break;
+        }
+    }
+    
+    if (index != -1) {
+        NSArray *pair = self.nativeAdlist[index];
+        NativeAdView *adView = pair[1];
+        
+        [adView loadNativeAdLayout:nativeAd];
+        
+        NSIndexPath *indexPath = [NSIndexPath indexPathForRow:index inSection:0];
+        [self.tableViewData addObject:adView]; // Add the adView to your data source.
+        [self.tableView insertRowsAtIndexPaths:@[indexPath] withRowAnimation:UITableViewRowAnimationAutomatic];
+        
+        // Scroll to the newly added cell
+        NSIndexPath *latestIndexPath = [NSIndexPath indexPathForRow:self.tableViewData.count - 1 inSection:0];
+        [self.tableView scrollToRowAtIndexPath:latestIndexPath atScrollPosition:UITableViewScrollPositionBottom animated:YES];
+    }
+}
+
+// ISNativeAdInteractionDelegate
+
+// This method gets invoked after a video has been clicked
+- (void)didClick:(nonnull LevelPlayNativeAd *)nativeAd withAdInfo:(nonnull ISAdInfo *)adInfo {
+    NSLog(@"%s",__PRETTY_FUNCTION__);
+}
+
+- (void)didRecordImpression:(nonnull LevelPlayNativeAd *)nativeAd withAdInfo:(nonnull ISAdInfo *)adInfo {
+    NSLog(@"%s",__PRETTY_FUNCTION__);
+}
+
+#pragma mark - UI TableView Delegate Functions
+- (nonnull UITableViewCell *)tableView:(nonnull UITableView *)tableView cellForRowAtIndexPath:(nonnull NSIndexPath *)indexPath {
+    id cellData = _tableViewData[indexPath.row];
```

Comment:
need to check for leaks here, we found some on the swift tester and fixed them.
please review https://github.com/ironsource-mobile/ios-sdk6/pull/1257/files

Reply:
not sure about this part, I did: 
cellData = nil;
please give me more review

Reply:
did you see the changes we did on the PR i sent you? we made them to your code that didn't clean the list of the native ads and to some more places

Reply:
hello, I double-checked the (void)nativeAdViewDeleteButtonTapped method in NativeAdViewController and emptied the ad object after use. 

I also ran the tool Instrument to check for leaks, please see attached. The highlighted part in the graph was when I clicked load, load, load, and load for 4 ads and then clicked delete 4 times. I don't see any native ad-related leaks. Please review. Thanks. 

![image](https://github.com/ironsource-mobile/Mediation-Demo-Apps/assets/109173079/6eebc2c6-8dba-46b2-a180-2a7725b316fa)